src/plugins/guided_onboarding/public/types.ts

```diff
@@ -15,18 +15,28 @@ import type {
   GuideConfig,
   GuideParams,
 } from '@kbn/guided-onboarding';
-import type { CloudStart } from '@kbn/cloud-plugin/public';
+import type { CloudSetup, CloudStart } from '@kbn/cloud-plugin/public';
+import { SharePluginStart } from '@kbn/share-plugin/public';
 import type { PluginStatus, PluginState } from '../common';
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface GuidedOnboardingPluginSetup {}
+export interface GuidedOnboardingPluginSetup {
+  cloud: CloudSetup;
+}
 
 export interface GuidedOnboardingPluginStart {
+  // getEsEndpointModal: (
+  //   core: CoreStart,
+  //   share: SharePluginStart,
+  //   cloud: CloudStart,
+  //   docLinks: DocLinksStart,
+  //   onClose?: () => {}
+  // ) => any;
   guidedOnboardingApi?: GuidedOnboardingApi;
 }
 
 export interface AppPluginStartDependencies {
-  cloud?: CloudStart;
+  cloud: CloudStart;
+  sharePluginStart: SharePluginStart;
```

Comment:
since this is provided by the Kibana framework, I think it should match the ID of the plugin

```suggestion
  share: SharePluginStart;
```